### PR TITLE
Add C# simple planner

### PR DESCRIPTION
**Pull request by dluc**
### Motivation and Context

Need for a simple planner showing how to leverage semantic functions to implement Intent Detection pattern and turn it into a planner, selecting a function given a goal.

The planner also allows to provide a Planning capability using less powerful models, like GPT3 (text-davinci-002) and GPT3.5 (text-davinci-003, GPT turbo 3.5), where often the more advanced planner can hallucinate.

Fix one bug in "WriterSkill/ShortPoem/skprompt.txt" leading to incorrect results when the goal was missing a termination (period or new line)

### Description

* New package `Microsoft.SemanticKernel.Planning.ActionPlanner`extending the Planning namespace. The package is automatically installed, ie it's part of SK metapackage.
* Move SequentialPlanner into the new package `Microsoft.SemanticKernel.Planning.SequentialPlanner` lib, part of the metapackage
* Add new Action Planner.
* Store planner prompts to skprompt.txt, using embedded resources to avoid files overwriting each other.

### CODE REVIEW

File: dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs
Comment (RogerBarreto) [inline dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs:1]: Any special reason why we are encouraging conflicts using the `Planner` class?

If I have an implementation using both `SequentialPlanner` and `BasePlanner` will need to enforce the name space... instead of having a dedicated class for it... ?

File: dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Skills.Planning.BasicPlanner;
```
Comment (shawncal) [inline dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs:7]: Planning is a top-level concept of SK. We do not want it buried under the "Skills." namespace.

There was a "PlannerSkill" and it was leading to confusion about how to interact with the, so it was decided by the devs responsible for planner that they'd remove it earlier this week. This just reintroduces the same problem.

**Review by shawncal — DISMISSED** [review]

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt [hunk 2]
```diff
@@ -1,11 +1,4 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-
-namespace Microsoft.SemanticKernel.Planning.Planners;
-
-internal static class SemanticFunctionConstants
-{
-    internal const string FunctionFlowFunctionDefinition =
-        @"Create an XML plan step by step, to satisfy the goal given.
+Create an XML plan step by step, to satisfy the goal given.
 To create a plan, follow these steps:
```
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt:2]: Remove the encoded `"` in the file.

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt:1]: > Sequential Planner requires GPT4. Introduce a planner compatible with GPT3/3.5.

Not sure what is meant here. Sequential Planner and its tests run again GPT3/3.5 already.

File: dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <!-- THIS PROPERTY GROUP MUST COME FIRST -->
+    <AssemblyName>Microsoft.SemanticKernel.Skills.Planning</AssemblyName>
```
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj:5]:
```Suggestion
    <AssemblyName>Microsoft.SemanticKernel.Planning</AssemblyName>
```

File: dotnet/src/SemanticKernel.Abstractions/Orchestration/PlanningException.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 using System;
 using Microsoft.SemanticKernel.Diagnostics;
 
-namespace Microsoft.SemanticKernel.Planning;
+namespace Microsoft.SemanticKernel.Orchestration;
 
 /// <summary>
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Abstractions/Orchestration/PlanningException.cs:8]: I'd expect this to stay in the package it's used in and the `planning` namespace.

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/PlannerConfig.cs [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.SemanticKernel.Planning.Planners;
+namespace Microsoft.SemanticKernel.Skills.Planning.SequentialPlanner;
 
 /// <summary>
 /// Common configuration for planner instances.
```
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Planning/SequentialPlanner/PlannerConfig.cs:8]: I think this or parts of it ought to be common.

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs [hunk 6]
```diff
@@ -1,36 +1,46 @@
...
 using Microsoft.SemanticKernel.Diagnostics;
 using Microsoft.SemanticKernel.Orchestration;
 
-namespace Microsoft.SemanticKernel.Planning.Planners;
+namespace Microsoft.SemanticKernel.Skills.Planning.SequentialPlanner;
 
 /// <summary>
 /// A planner that uses semantic function to create a sequential plan.
 /// </summary>
 public class SequentialPlanner
```
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs:13]: I think we should leave this in the core and any other planners we create/experiment with go in the Microsoft.SemanticKernel.Planning package.

File: dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs [hunk 7]
```diff
@@ -57,7 +57,10 @@ public void CanCallToPlanFromXml()
 </plan>";
 
         // Act
+        // TODO - a test should not rely on the code under test, e.g. if there's a bug in ToPlanFromXml
+        // it'll result in false positives/negatives
         var plan = planString.ToPlanFromXml(kernel.CreateNewContext());
```
Comment (lemillermicrosoft) [inline dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs:62]: `ToPlanFromXml` is the code being tested though. I don't understand the note.

File: dotnet/src/Skills/Skills.UnitTests/Planning/SequentialPlanner/PlanParserTests.cs [hunk 8]
```diff
@@ -132,6 +132,7 @@ public void CanCallToPlanFromXml()
 </plan>";
 
         // Act
+        // TODO: unit tests should not depend on the code under test
```
Comment (lemillermicrosoft) [inline dotnet/src/Skills/Skills.UnitTests/Planning/SequentialPlanner/PlanParserTests.cs:135]: Same as other, not sure what the concern here is.

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs (hunk 6 above)
Comment (RogerBarreto) [reply dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs:13]: @lemillermicrosoft if we are encouraging new planners to be created I would do the same approach that was done for the Services abstractions:

Example:

Having a: `SemanticKernel.Abstractions.Planning`
and `SemanticKernel.Planning.SequentialPlanner` depending on it.

Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs:13]: Yeah originally I had an `IPlanner` interface but was encouraged to remove it. What do you think would be appropriate to have in `SemanticKernel.Abstractions.Planning` in this case? `PlanException` maybe? Anything else?

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/SequentialPlanner/skprompt.txt:1]: we have received several complaints that the sequential planner hallucinates when not using GPT4

File: dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj (hunk 3 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj:5]: the planner is a skill, I would follow the existing naming convention

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/PlannerConfig.cs (hunk 5 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/SequentialPlanner/PlannerConfig.cs:8]: the concept of included/excluded functions, relevance and maxrelevance is a feature of the sequential planner, I don't think we should assume that every planner will use it. The basic planner for example doesn't use any of them

File: dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs (hunk 6 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/SequentialPlanner/SequentialPlanner.cs:13]: we can revisit the location but I would not split the planners in separate locations. The basic planner and a bunch of code that is coming need to be in the same place. Using a separate library makes it easier to support backward compatibility of the core kernel. 
If we place planners in the core, after we promote the nuget to beta or v1, it will be much harder making any change.

File: dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs (hunk 7 above)
Comment (dluc) [reply dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs:62]: is a best practice in unit testing not relying on the code under test, even if it's another class. I think the main problem here is that there are many classes coupled together and there is no way to inject mocks, but fixing that is a big tasks.

Comment (lemillermicrosoft) [reply dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs:62]: Which classes are coupled in this instance?

File: dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj (hunk 3 above)
Comment (lemillermicrosoft) [reply dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj:5]: To me it's more of a SkillGenerator or FunctionGenerator. It could be wrapped in a skill but I don't think we should be doing that right now. I'd like to understand the scenario more if we think urgent for build.

File: dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs (hunk 7 above)
Comment (dluc) [reply dotnet/src/IntegrationTests/Planning/SequentialPlanner/SequentialPlanParserTests.cs:62]: probably better to discuss offline in front of the IDE, feel free to reach out

File: dotnet/src/SemanticKernel.Abstractions/Orchestration/PlanningException.cs (hunk 4 above)
Comment (dluc) [reply dotnet/src/SemanticKernel.Abstractions/Orchestration/PlanningException.cs:8]: I'm going back to previous design conversations here: plans are functions compositions, so `Plan` and `PlanningException` are in the same category of ISKfunction, ContextVariables, etc.

File: dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj (hunk 3 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/Skills.Planning.csproj:5]: This goes back to the design and message we've been giving since the start: planners and memory are skills. For instance we will have planner skills discovering other planners. These building blocks can be combined in many ways to achieve higher level concepts (like function generator) but they are still skills and can be composed on many ways. If we decide that they are something else, then the design will diverge and they will become unusable in scenarios where one wants to compose skills including these planners.

File: dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs (hunk 1 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs]: Namespacing and libraries do not bury the code. We lead users with examples and documentation to show which components and features to leverage. Otherwise we would have to put everything in one folder and the code would be unmaintainable.
I'm just going to ignore the "*the responsible devs*" part, which is out of scope.

File: dotnet/src/Skills/Skills.UnitTests/Planning/SequentialPlanner/PlanParserTests.cs (hunk 8 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.UnitTests/Planning/SequentialPlanner/PlanParserTests.cs:135]: see other comment, easier to discuss offline using IDE

File: dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs (hunk 1 above)
Comment (dluc) [reply dotnet/src/Skills/Skills.Planning/BasicPlanner/Planner.cs]: moved to Extension package

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+
+        return list.ToString();
+    }
+
+    // TODO: generate string programmatically
+    // TODO: use goal to find relevant examples
+    [SKFunction("List a few good examples of plans to generate")]
+    [SKFunctionName("GoodExamples")]
+    [SKFunctionInput(Description = "The current goal processed by the planner", DefaultValue = "")]
+    public string GoodExamples(string goal, SKContext context)
```
Comment (shawncal) [inline dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:148]: Why an SKFunction? Is the AI meant to call this?

File: dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+{{basicPlanner.GoodExamples}}
+{{basicPlanner.EdgeCaseExamples}}
+[END OF EXAMPLES]
+[REAL SCENARIO STARTS HERE]
+- List of functions:
+{{basicPlanner.ListOfFunctions}}
+- End list of functions.
+Goal: {{ $input }}
+{"plan":{
+"rationale":
```
Comment (shawncal) [inline dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt:12]: Do we want it to generate rationale every time, or could this be a "debug mode" optional feature? Just thinking that, if this really is the most basic, cheap planner, minimizing token consumption is important as it could be called many, many times, including in tight loops.

File: dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
+A planner takes a list of functions, a goal, and chooses which function to use.
+For each function the list includes details about the input parameters.
+{{basicPlanner.GoodExamples}}
+{{basicPlanner.EdgeCaseExamples}}
+[END OF EXAMPLES]
```
Comment (shawncal) [inline dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt:5]: Are there always some examples? Or could this [END] be left hanging?

File: dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs [hunk 12]
```diff
@@ -6,15 +6,13 @@
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Microsoft.SemanticKernel.Memory;
-using Microsoft.SemanticKernel.Planning.Planners;
+using Microsoft.SemanticKernel.Planning.Sequential;
 using Microsoft.SemanticKernel.SkillDefinition;
 
-#pragma warning disable IDE0130
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs:12]: Please leave these in and add elsewhere as appropriate.

File: dotnet/src/SemanticKernel.MetaPackage/SemanticKernel.MetaPackage.csproj [hunk 13]
```diff
@@ -16,8 +16,10 @@ Empowers app owners to integrate cutting-edge LLM technology quickly and easily
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\SemanticKernel\SemanticKernel.csproj" />
-    <ProjectReference Include="..\Connectors\Connectors.AI.OpenAI\Connectors.AI.OpenAI.csproj" PrivateAssets="none"/>
+    <ProjectReference Include="..\SemanticKernel\SemanticKernel.csproj" PrivateAssets="none" />
+    <ProjectReference Include="..\Connectors\Connectors.AI.OpenAI\Connectors.AI.OpenAI.csproj" PrivateAssets="none" />
+    <ProjectReference Include="..\Extensions\Planning.ActionPlanner\Planning.ActionPlanner.csproj" PrivateAssets="none" />
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.MetaPackage/SemanticKernel.MetaPackage.csproj:21]: Nit: Maybe just `Planning\ActionPlanner\Planning.ActionPlanner.csproj` to follow the pattern with Skills.

Comment (lemillermicrosoft) [reply dotnet/src/SemanticKernel.MetaPackage/SemanticKernel.MetaPackage.csproj:21]: I also know we've been saying `Planning` the whole time, but maybe `Planners` makes more sense if that's all we think we will have in that namespace are various planners.

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs (hunk 9 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:148]: yes see the prompt

File: dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt (hunk 10 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt:12]: this is a request we've had from users, the planner explain why it's making some choices. It's also a common feature in other frameworks

File: dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt (hunk 11 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt:5]: in my experience this is clearer and works well with old models, ie clearly marking each section start and end by name

File: dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs (hunk 12 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs:12]: what are they for?

Comment (dluc) [reply dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs:12]: I don't see any build warning after removing them

File: dotnet/src/SemanticKernel.MetaPackage/SemanticKernel.MetaPackage.csproj (hunk 13 above)
Comment (dluc) [reply dotnet/src/SemanticKernel.MetaPackage/SemanticKernel.MetaPackage.csproj:21]: I don't want to have an explosion of folders for everything we extend, in fact if we don't ship planners as skills, I would move skills under the extensions folder. For now I find a generic folder better scales to allow other extensions

**Review by lemillermicrosoft — CHANGES_REQUESTED** [review]

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Planning.Action;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+// ReSharper disable once CheckNamespace - Using NS of Plan
+namespace Microsoft.SemanticKernel.Planning;
+
+public sealed class ActionPlanner
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:17]: Documentation

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+            maxTokens: 1024,
+            stopSequences: new[] { StopSequence });
+
+        kernel.ImportSkill(this, skillName: SkillName);
+
+        this._kernel = kernel;
+        this._context = kernel.CreateNewContext();
+    }
+
+    public async Task<Plan> CreatePlanAsync(string goal)
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:55]: Planning on adding unit or integration tests in this PR or a follow up?

**Review by lemillermicrosoft — CHANGES_REQUESTED** [review]

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs (hunk 15 above)
Comment (lemillermicrosoft) [reply dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:55]: https://github.com/orgs/microsoft/projects/811/views/1?pane=issue&itemId=26713338

File: dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs (hunk 12 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.SequentialPlanner/SKContextSequentialPlannerExtensions.cs:12]: added everywhere I could find it - didn't know about the feature, nice one

File: dotnet/src/Extensions/Extensions.UnitTests/Extensions.UnitTests.csproj [hunk 16]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <AssemblyName>SemanticKernel.Extensions.UnitTests</AssemblyName>
+    <RootNamespace>SemanticKernel.Extensions.UnitTests</RootNamespace>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>disable</ImplicitUsings>
+    <IsPackable>false</IsPackable>
+    <NoWarn>CA2007,VSTHRD111</NoWarn>
```
[the remark this answers is not in the document]
Comment (dluc) [reply dotnet/src/Extensions/Extensions.UnitTests/Extensions.UnitTests.csproj:10]: done

File: dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+                list.AppendLine($"// {AddPeriod(func.Description)}");
+
+                // Function name
+                list.AppendLine($"{func.SkillName}.{func.Name}");
+
+                // Function parameters
+                foreach (var p in func.Parameters)
+                {
+                    var description = string.IsNullOrEmpty(p.Description) ? p.Name : p.Description;
+                    list.AppendLine($"Parameter \"{p.Name}\": {AddPeriod(description)}");
```
[the remark this answers is not in the document]
Comment (dluc) [reply dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs:231]: https://github.com/orgs/microsoft/projects/811/views/1?pane=issue&itemId=26715228

File: dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+{
+    /// <summary>
+    /// Create a manual-friendly string for a function.
+    /// </summary>
+    /// <param name="function">The function to create a manual-friendly string for.</param>
+    /// <returns>A manual-friendly string for a function.</returns>
+    internal static string ToManualString(this FunctionView function)
+    {
+        var inputs = string.Join("\n", function.Parameters.Select(p => $"    - {p.Name}: {p.Description}"));
+        return $"  {function.ToFullyQualifiedName()}:\n    description: {function.Description}\n    inputs:\n{inputs}";
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs:18]: Some merge changes were missed here -- I'd expect to see the old file deleted or marked a rename. I can resolve in follow up, too.

File: dotnet/src/SemanticKernel/SkillDefinition/ImportSemanticSkillFromDirectory.cs [hunk 19]
```diff
@@ -4,11 +4,12 @@
 using System.IO;
 using Microsoft.Extensions.Logging;
 using Microsoft.SemanticKernel.Diagnostics;
-using Microsoft.SemanticKernel.Orchestration;
 using Microsoft.SemanticKernel.SemanticFunctions;
+using Microsoft.SemanticKernel.SkillDefinition;
 using Microsoft.SemanticKernel.Text;
 
-namespace Microsoft.SemanticKernel.KernelExtensions;
+// ReSharper disable once CheckNamespace - Using the namespace of IKernel
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel/SkillDefinition/ImportSemanticSkillFromDirectory.cs:11]: nit: missed pragma. Can handle in follow up.

File: dotnet/src/SemanticKernel/SkillDefinition/InlineFunctionsDefinitionExtension.cs [hunk 20]
```diff
@@ -4,10 +4,10 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.SemanticKernel.Diagnostics;
-using Microsoft.SemanticKernel.Orchestration;
 using Microsoft.SemanticKernel.SemanticFunctions;
+using Microsoft.SemanticKernel.SkillDefinition;
 
-// ReSharper disable once CheckNamespace // Extension methods
+// ReSharper disable once CheckNamespace - Using the namespace of IKernel
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel/SkillDefinition/InlineFunctionsDefinitionExtension.cs:10]: nit: missed pragma, can address in follow up.

File: dotnet/src/SemanticKernel/SkillDefinition/SkillCollection.cs [hunk 21]
```diff
@@ -15,8 +15,8 @@ namespace Microsoft.SemanticKernel.SkillDefinition;
 /// The class holds a list of all the functions, native and semantic, known to the kernel instance.
 /// The list is used by the planner and when executing pipelines of function compositions.
 /// </summary>
-[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = "It is a collection")]
-[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1711:Identifiers should not have incorrect suffix", Justification = "It is a collection")]
+[SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = "It is a collection")]
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel/SkillDefinition/SkillCollection.cs:18]: nit: is CA170 required? I saw in other review that it wasn't needed. Unrelated, feel free to resolve.

**Review by lemillermicrosoft — APPROVED** [review]
LGTM. Left some nit comments. All can be resolved as appropriate.

File: dotnet/src/SemanticKernel/SkillDefinition/SkillCollection.cs (hunk 21 above)
Comment (dluc) [reply dotnet/src/SemanticKernel/SkillDefinition/SkillCollection.cs:18]: I've been getting both warnings, very confusing, so I added both (this is old code)

File: dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs (hunk 18 above)
Comment (dluc) [reply dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs]: I re-did the "git mv", the file should have the right content

File: dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SequentialPlanParser.cs
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SequentialPlanParser.cs:1]: Something went wrong here, this isn't a TestHelper.

File: dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/ExceptionExtensions.cs
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/ExceptionExtensions.cs:1]: Was this file intended?

File: dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/FunctionViewExtensions.cs
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/FunctionViewExtensions.cs:1]: Ah here's part of the problem with the file move. Not a test helper.

File: dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SKContextExtensions.cs
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SKContextExtensions.cs:1]: Another testhelper issue.

**Review by lemillermicrosoft — CHANGES_REQUESTED** [review]

**Review by lemillermicrosoft — APPROVED** [review]

File: dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SequentialPlanParser.cs
Comment (dluc) [reply dotnet/src/Extensions/Extensions.UnitTests/Planning/SequentialPlanner/TestHelpers/SequentialPlanParser.cs:1]: I removed the test helpers part. I think the sequential planner unit tests should be refactored not to depend on code in other libraries, particularly not on code closely related to the code under test. The major problem with the current approach is some tests can pass even if there are bugs, because bugged code can be used to test bugged code. I'll leave that to some other PR to address
